virttest/qemu_devices/qdevices.py

```diff
@@ -907,23 +907,57 @@ def hotplug_qmp(self):
         if params.get("host-nodes"):
             host_nodes = list(map(int, params["host-nodes"].split()))
             params["host-nodes"] = host_nodes
+        for k in params:
+            params[k] = True if params[k] == "yes" else params[k]
+            params[k] = False if params[k] == "no" else params[k]
         kwargs = {"qom-type": backend,
                   "id": params.pop("id"),
                   "props": dict(params)}
         return "object-add", kwargs
 
     def verify_unplug(self, out, monitor):
-        out = monitor.info("memory-devices", debug=False)
-        memdev = "/object/%s" % self.get_qid()
-        if memdev not in out:
-            return True
-        return False
+        """
+        :param out: Output of the unplug command
+        :param monitor: Monitor used for unplug
+        :return: True when successful, False when unsuccessful
+        """
+        out = monitor.info("memdev", debug=False)
+        memdev = self.get_qid()
+        for dev in out:
+            if dev["id"] == memdev:
+                return False
+        return True
 
     def verify_hotplug(self, out, monitor):
-        out = monitor.info("memory-devices", debug=False)
-        memdev = "/object/%s" % self.get_qid()
-        if memdev in out:
-            return True
+        """
+        :param out: Output of the hotplug command
+        :param monitor: Monitor used for hotplug
+        :return: True when successful, False when unsuccessful
+        """
+        out = monitor.info("memdev", debug=False)
+        memdev = self.get_qid()
+        params = self.params.copy()
+        for dev in out:
+            if dev["id"] == memdev:
+                if not params.get("host-nodes") and len(dev["host-nodes"]):
+                    return False
+                if params.get("host-nodes"):
+                    host_nodes = list(map(int, params["host-nodes"].split()))
+                    if dev["host-nodes"] != host_nodes:
```

Comment:
if dev["host-nodes"] is single element then it is better to check as,
```python
if dev["host-nodes"] in host_nodes:
```
or if dev["host-nodes"] is a list, then it is better to check as,
```python
if dev["host-nodes"].sort() != host_nodes.sort():
```